src/files-action.js

```diff
@@ -20,25 +21,47 @@
  *
  */
 import { generateUrl } from '@nextcloud/router'
+import { translate as t } from '@nextcloud/l10n'
+import { registerFileAction, FileAction, Permission, DefaultType } from '@nextcloud/files'
+/* eslint-disable-next-line import/no-unresolved */
+import ContactSvg from '@mdi/svg/svg/account-multiple.svg?raw'
 
 const mime = 'text/vcard'
 const name = 'contacts-import'
+const nextcloudVersionIsGreaterThanOr28 = parseInt(OC.config.version.split('.')[0]) >= 28
 
-window.addEventListener('DOMContentLoaded', () => {
-	if (OCA.Files && OCA.Files.fileActions) {
-		OCA.Files.fileActions.registerAction({
-			name,
-			displayName: t('contacts', 'Import'),
-			mime,
-			permissions: OC.PERMISSION_READ,
-			iconClass: 'icon-contacts-dark',
-			actionHandler(fileName, context) {
-				const absPath = `${context.dir === '/' ? '' : context.dir}/${fileName}`
-				window.location = generateUrl(`/apps/contacts/import?file=${absPath}`)
-			},
-		})
-		OCA.Files.fileActions.setDefault(mime, name)
-		return
-	}
-	console.error('Unable to register vcf import action')
-})
+if (nextcloudVersionIsGreaterThanOr28) {
+	registerFileAction(new FileAction({
+		id: name,
+		displayName: () => t('contacts', 'Import'),
+		default: DefaultType.DEFAULT,
+		mime,
+		enabled: (nodes) => {
+			return nodes.every((node) => node.mime === mime && (node.permissions & Permission.READ))
```

Comment:
Using `some` with the opposite check is faster as it fill stop at the first false return
Something like 
``` suggestion
			return nodes.some((node) => node.mime !== mime || (node.permissions & Permission.READ) === 0)
```